src/main/java/bio/terra/service/tabulardata/google/BigQueryPdao.java

```diff
@@ -288,14 +310,57 @@ public void createSnapshot(Snapshot snapshot, List<String> rowIds) throws Interr
         List<WalkRelationship> walkRelationships = WalkRelationship.ofAssetSpecification(asset);
         walkRelationships(datasetBqDatasetName, snapshotName, walkRelationships, rootTableId, projectId, bigQuery);
 
-        // create the views
-        List<String> bqTableNames = createViews(datasetBqDatasetName, snapshotName, snapshot, projectId, bigQuery);
+        snapshotViewCreation(datasetBqDatasetName, snapshotName, snapshot, projectId, bigQuery, bigQueryProject);
+    }
 
-        // set authorization on views
-        List<Acl> acls = convertToViewAcls(projectId, snapshotName, bqTableNames);
-        bigQueryProject.addDatasetAcls(datasetBqDatasetName, acls);
+    private static final String getAllLiveViewDataTemplate =
+        "INSERT INTO `<project>.<snapshot>.<dataRepoTable>` " +
+            "(<dataRepoTableId>, <dataRepoRowId>) " + // TODO is this line necessary?
+            "SELECT <dataRepoRowId>, '<tableId>' FROM `<project>.<dataset>.<liveView>`";
+
+    public void createSnapshotWithLiveViews(
+        Snapshot snapshot,
+        Dataset dataset) throws InterruptedException {
+
+        BigQueryProject bigQueryProject = bigQueryProjectForSnapshot(snapshot);
+        String projectId = bigQueryProject.getProjectId();
+        String snapshotName = snapshot.getName();
+        BigQuery bigQuery = bigQueryProject.getBigQuery();
+
+        String datasetBqDatasetName = prefixName(dataset.getName());
+
+        // create snapshot BQ dataset
+        snapshotCreatBQDataset(bigQueryProject, snapshot);
+
+        // create the row id table (row id col and table id col)
+        bigQueryProject.createTable(snapshotName, PDAO_ROW_ID_TABLE, rowIdTableSchema());
+
+        // get source dataset table live views
+        List<DatasetTable> tables = dataset.getTables();
+
+        // create a snapshot table based on the live view data row ids
+        for (DatasetTable table : tables) {
+            TableId liveViewId = TableId.of(datasetBqDatasetName, table.getName());
+            String liveViewTableName = liveViewId.getTable();
+            // for each live view, get a list of all included row ids
+
+            ST sqlTemplate = new ST(getAllLiveViewDataTemplate);
+            sqlTemplate.add("project", projectId);
+            sqlTemplate.add("snapshot", snapshotName);
+            sqlTemplate.add("dataRepoTable", PDAO_ROW_ID_TABLE);
+            sqlTemplate.add("dataRepoTableId", PDAO_TABLE_ID_COLUMN);
+            sqlTemplate.add("dataRepoRowId", PDAO_ROW_ID_COLUMN);
+            sqlTemplate.add("tableId", table.getId());
+            sqlTemplate.add("dataset", datasetBqDatasetName);
+            sqlTemplate.add("liveView", liveViewTableName);
+        }
+
+        // TODO there is not enough validation - thought other than dataset existing, not sure what would be ideal
+
+        snapshotViewCreation(datasetBqDatasetName, snapshotName, snapshot, projectId, bigQuery, bigQueryProject);
```

Comment:
You won't be able to use that view creation logic. These views will not have a rowid table or a join. They should be something like:
```
"SELECT " + PDAO_ROW_ID_COLUMN + ", <columns; separator=\",\"> FROM (" +
            "SELECT S." + PDAO_ROW_ID_COLUMN + ", <mappedColumns; separator=\",\"> " +
            "FROM <project>.<dataset>.<mapTable> S";
```

Reply:
Something to watch out for if re-implementing pieces of the view generation: Make sure the `mapTable` referenced in the view is the _raw_ table, not the live view name. There's a separate getter on `DatasetTable` that will give you the value.

Reply:
@danxmoran Is this not correct?

`            sqlTemplate.add("mapTable", mapTable.getFromTable().getRawTableName());
`

Reply:
That is correct, I was just responding to Dan’s comment about potentially not reusing the existing view-creation logic

Reply:
cool--thank you!